main.py

```diff
@@ -0,0 +1,241 @@
+    temperature record (temp_record) and make the dict
+    
+    """    
+    def readDataFromSingleFile(self, month):
+        try :
+            with open(f'weather_reports/Murree_weather_{self.year}_{month}.txt', 'r') as file:
+                single_file_records = {}
+                single_line = file.readline()
+                _, *col_names = single_line.split(',')
+                single_date_record = {}
+                
+                while True:
+                    line = file.readline()
+                    if line:
+                        date, *temp_records = line.split(',')
+                        for index in range(len(temp_records)):
+                            single_date_record[col_names[index]] = temp_records[index]
+                            
+                        single_file_records.update({date: single_date_record})
+                        single_date_record = {}  
+                    else:
+                        break
+                self.data.append(single_file_records)
+        except : 
+            pass
+            
+        
+class CalculateTemperatureValues:
+    def __init__(self, file_obj):
+        self.data = file_obj
+        
+    def calculateTemperature(self, type_of_calculation):
+        if type_of_calculation == 'H':
+            return self._calculateTemperatureWithField('Max TemperatureC', True)
+        elif type_of_calculation == 'L': 
+            return self._calculateTemperatureWithField('Min TemperatureC', False)
+        elif type_of_calculation == 'Hu': 
+            return self._calculateTemperatureWithField('Max Humidity', True)
+        elif type_of_calculation == 'avg_L':
+            return self._calculateTemperatureWithField('Mean TemperatureC', False)
+        elif type_of_calculation == 'avg_H':
+            return self._calculateTemperatureWithField('Mean TemperatureC', True)
+        elif type_of_calculation == 'avg_Hu':
+            return self._calculateTemperatureWithField('Max Humidity', True)
+        else:
+            raise Exception("Incorrect Input")
+    
+    def _calculateTemperatureWithField(self, field, maxNumber):
+        latest_data = ""
+        required_data = ""
+        for single_file_record in self.data.data:
+            latest_data = self._calculateInfoFromFile(single_file_record, field, maxNumber)
+            if(not required_data):
+                required_data = latest_data
+                
+            if(maxNumber and latest_data["temperature"] > required_data["temperature"]):
+                required_data = latest_data 
+            else:
+                if latest_data["temperature"] < required_data["temperature"]:
+                    required_data = latest_data 
+                    
+            print(required_data)
+               
+        return required_data
+        
+    def _calculateInfoFromFile(self, single_file_record, field, maxNumber):
```

Comment:
Can we make the classes and functions be independent of whether the data was loaded from file or db or any other source?